wscript

```diff
@@ -67,21 +67,27 @@ def options(opt):
 
     opt.recurse('cfitsio')
 
-def _setup_openmp(conf):
-    conf.start_msg('Checking for OpenMP')
-
-    if conf.options.disable_openmp:
-        conf.end_msg('OpenMP disabled.', 'YELLOW')
-        return
 
-    try:
-        conf.check_cc(
-            header_name="omp.h", lib="gomp", cflags="-fopenmp",
-            uselib_store="OPENMP")
-    except Errors.ConfigurationError:
-        conf.end_msg("OpenMP not found.", 'YELLOW')
+def _setup_openmp(conf):
+    """
+    Detects openmp flags and sets the OPENMP ``CFLAGS``/``LINKFLAGS``
+    """
+    for x in ('-fopenmp','-openmp','-mp','-xopenmp','-omp','-qsmp=omp'):
+        try:
+            conf.check_cc(
+                msg = 'Checking for OpenMP: {0}'.format(x),
+                fragment = '''#include <omp.h>\nint main() { return omp_get_num_threads(); }''',
+                cflags = x,
+                linkflags = (x, '-ldl', '-lpthread', '-lrt'),
```

Comment:
Why is pthreads needed? Especially since this change is to allow for different implementations of openMP. What happens on Windows?